### PR TITLE
add seed liquidity based on ilm curve

### PR DESCRIPTION
**Pull request by codewithgun**

### CODE REVIEW

File: ts-client/src/dlmm/index.ts [hunk 1]
```diff
@@ -3623,6 +3629,339 @@ export class DLMM {
...
+    owner: PublicKey,
+    operator: PublicKey,
+    feeOwner: PublicKey,
+    seedAmount: BN,
+    curvature: number,
+    minPrice: number,
+    maxPrice: number,
+    base: PublicKey
+  ): Promise<TransactionInstruction[][]> {
+    const toLamportMultiplier = new Decimal(
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:3647]: Can we write a function to calculate amounts for each bin? so easier for unit test

File: ts-client/src/dlmm/index.ts [hunk 2]
```diff
@@ -3623,6 +3629,339 @@ export class DLMM {
...
+
+      if (amount.gt(maxAmount)) {
+        maxAmount = amount;
+      }
+
+      binDepositAmount.set(i, amount);
+    }
+
+    // Look for multiplier with least loss
+    let decompressMultiplier = new BN(10);
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:3714]: I thought we just go with min as 10^token_decimals?

File: ts-client/src/dlmm/index.ts (hunk 1 above)
Comment (codewithgun) [reply ts-client/src/dlmm/index.ts:3647]: Updated

File: ts-client/src/dlmm/index.ts (hunk 2 above)
Comment (codewithgun) [reply ts-client/src/dlmm/index.ts:3714]: Fixed, seems like this is more simplier.

**Review by andrewsource147 — APPROVED** [review]

**Pull request merged**